src/lib/src/grid3d/corners.cpp

```diff
@@ -0,0 +1,102 @@
+#include <cstddef>
+#include <vector>
+
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+
+#include <xtgeo/numerics.hpp>
+#include <xtgeo/xtgeo.h>
+
+namespace py = pybind11;
+
+namespace XTGeo::Grid3D {
+
+/*
+ * Given a cell coordinate (i, j, k), find all corner coordinates as an
+ * array with 24 values.
+ *
+ *      Top  --> i-dir     Base c
+ *
+ * (6,7,8) (9,10,11) (18,19,20) (21,22,23)
+ *    |-------|          |-------|
+ *    |       |          |       |
+ *    |       |          |       |
+ *    |-------|          |-------|
+ * (0,1,2) (3,4,5)   (12,13,14) (15,16,17)
+ * (x,y,z)
+ *
+ * @param i The (i) coordinate
+ * @param j The (j) coordinate
+ * @param k The (k) coordinate
+ * @param ncol The grid column/nx dimension
+ * @param nrow The grid row/ny dimension
+ * @param nlay The grid layer/nz dimension
+ * @param coordsv Grid coordnates vector
+ * @param zcornsv Grid Z corners vector
+ * @return vector of 24 doubles with the corner coordinates.
+ */
+std::vector<double>
```

Comment:
Returning a `std::vector` to Python will copy all the contents to a list. It's fine in this case, since it's only 24 values.